src/wp-includes/media.php

```diff
@@ -1956,6 +1956,31 @@ function wp_image_use_alternate_mime_types( $image, $context, $attachment_id ) {
 				break;
 			}
 
+			/**
+			 * Filters the image HTML tag.
+			 *
+			 * This filter can be used to replace target mime_type image filename for different sizes
+			 * in the generated <img> HTML tag.
+			 *
+			 * @since 6.1.0
+			 *
+			 * @param array  $image         The HTML `img` tag.
+			 * @param string $attachment_id The attachment ID.
+			 * @param string $metadata      Image size metadata.
+			 * @param string $name          Size identifier - e.g. 'full', 'medium', 'small' etc.
+			 * @param string $target_mime   Image mime type.
+			 * @param string $context       Additional context, like the current filter name or the function name from where this was called.
+			 *
+			 * @return string Updated <img> HTML tag.
+			 */
+			$filtered_image = apply_filters( 'wp_content_pre_replace_additional_image_source', $image, $attachment_id, $metadata, 'full', $target_mime, $context );
+
+			if ( $filtered_image !== $image ) {
```

Comment:
can you add a doc block explaining what this check is for/what the code inside does?

Reply:
@adamsilverstein Can the below doc block make sense?

`// If the filtered image is not the same as the image, then use the filtered image instead of the original image.`

Reply:
👍🏼 sure, that sounds good, maybe use that to document the return value of the filter.